internal/internal_task_handlers.go

```diff
@@ -1441,6 +1442,17 @@ func isCommandMatchEvent(d *commandpb.Command, e *historypb.HistoryEvent, strict
 			}
 		}
 		return true
+
+	case enumspb.COMMAND_TYPE_MODIFY_WORKFLOW_PROPERTIES:
+		if e.GetEventType() != enumspb.EVENT_TYPE_WORKFLOW_PROPERTIES_MODIFIED {
+			return false
+		}
+		eventAttributes := e.GetWorkflowPropertiesModifiedEventAttributes()
+		commandAttributes := d.GetModifyWorkflowPropertiesCommandAttributes()
+		if strictMode && !isMemoMatched(eventAttributes.UpsertedMemo, commandAttributes.UpsertedMemo) {
+			return false
+		}
```

Comment:
StrictMode was there for historical reason. But I don't think they are needed anymore. We probably should remove strictMode. @cretz 
(No change requested for this PR.)